zulipterminal/ui_tools/boxes.py

```diff
@@ -580,11 +580,21 @@ def soup2markup(self, soup: Any, **state: Any) -> List[Any]:
                     for data in [link, text]
                 ]   # Split on '://' is for cases where text == link.
                 if link_without_scheme == text_without_scheme:
-                    segment = text.split('/')[-1]
-                    # Replace text with its last segment if the segment has
-                    # something significant than simply the 'domain name'.
-                    if segment != text_without_scheme:
-                        text = segment
+                    last_segment = text.split('/')[-1]
+                    if '.' in last_segment:
+                        new_text = last_segment  # Filename.
+                    elif text.startswith(self.model.server_url):
+                        # Relative URL.
+                        new_text = text.split(self.model.server_url)[-1]
+                    else:
+                        new_text = (
+                            parsed_link.netloc if parsed_link.netloc
+                            else text.split('/')[0]
+                        )  # Domain name.
+                    # Replace text with new_text if new_text is different from
+                    # text_without_scheme.
```

Comment:
This comment feels redundant now. Should I remove it?

Reply:
Agreed. The actual code is pretty simple to follow here.